pkg/pillar/cmd/zedmanager/zedmanager.go

```diff
@@ -897,6 +945,14 @@ func handleZedAgentStatusImpl(ctxArg interface{}, key string,
 	statusArg interface{}) {
 	ctxPtr := ctxArg.(*zedmanagerContext)
 	status := statusArg.(types.ZedAgentStatus)
+	if status.ConfigGetStatus == types.ConfigGetSuccess {
+		log.Warnf("@ohm: Got the ConfigGetSuccess from ZedAgent (key %s), set the delay time base to now!", key)
+		ctxPtr.delayFromTime = time.Now()
+		if status.DomainConfigsNum == 0 {
+			log.Warnf("@ohm: No VMs to start, close the delay window")
+			ctxPtr.delayWindowPassed = true
+		}
```

Comment:
According to Gopi we should not apply the delays at all if at the start of the device there were no predefined VMs.

Reply:
I think that is a mistake or confusion.
If the device was booted a week ago and then you deploy an app instance with a 10 minute delay, then that delay has already passed.
But if the device was booted 10 minutes ago and you deploy an app instance with a 1 hour delay then I think it is perfectly reasonable to have that wait until 1 hour passed since boot,

The key thing is that we always want to have the delay be relative to the boot time (as defined by when it processed the config) and not be based on when the application instance is created.

We want the mechanism to be simple - to explain and to implement.